taxonomy/utils.py

```diff
@@ -238,3 +242,83 @@ def get_course_jobs(course_key):
             }
         )
     return data
+
+
+def get_translated_course_description(course_key, course_description):
+    """
+    Return translated course description.
+
+    Create translation for course description if translation object doesn't already exist.
+     OR update translation if course description changed from previous description in translation and
+      return the translated course description.
+
+    Arguments:
+        course_key (str): Key of the course whose description needs to be translated.
+        course_description (str): Full course description of the course which needs to be translated.
+
+    Returns:
+        str: Translated full course description.
+    """
+    translation = Translation.objects.filter(source_model_name='Course',
+                                             source_model_field='full_description',
+                                             source_record_identifier=course_key).first()
+    if translation:
```

Comment:
you can combine these 2 if conditions like

```
if translation and translation.source_text != course_description:
```

Reply:
No, we can't do it this way. Because we need to return `translated_text` of `translation` object if it exists already. We need new translation of `course_description` if it changes to previous one otherwise return the old translation